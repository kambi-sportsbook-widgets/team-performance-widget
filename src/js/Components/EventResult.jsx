import React from 'react';

const EventResult = ({ homeName, homeScore, awayName, awayScore }) => {
   let homeClasses = 'kw-eventresult__participant';
   let awayClasses = 'kw-eventresult__participant';
   if (homeScore > awayScore) {
      homeClasses += ' KambiWidget-card-text-color';
      awayClasses += ' KambiWidget-card-support-text-color';
   } else if (awayScore > homeScore) {
      homeClasses += ' KambiWidget-card-support-text-color';
      awayClasses += ' KambiWidget-card-text-color';
   } else {
      homeClasses += ' KambiWidget-card-support-text-color';
      awayClasses += ' KambiWidget-card-support-text-color';
   }
   return (
      <div className='kw-eventresult'>
         <div className={homeClasses}>
            <div className='kw-eventresult__participant__name'>{homeName}</div>
            <div className='kw-eventresult__participant__score'>{homeScore}</div>
         </div>
         <div className={awayClasses}>
            <div className='kw-eventresult__participant__name'>{awayName}</div>
            <div className='kw-eventresult__participant__score'>{awayScore}</div>
         </div>
      </div>
   );
};

EventResult.propTypes = {
   /**
    * Home participant name
    */
   homeName: React.PropTypes.string.isRequired,

   /**
    * Home participant score
    */
   homeScore: React.PropTypes.number.isRequired,

   /**
    * Away participant name
    */
   awayName: React.PropTypes.string.isRequired,

   /**
    * Away participant score
    */
   awayScore: React.PropTypes.number.isRequired
};

export default EventResult;
