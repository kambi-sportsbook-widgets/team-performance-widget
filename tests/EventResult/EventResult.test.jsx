/* eslint-env jest */
import React from 'react';
import EventResult from '../../src/js/Components/EventResult';
import ReactTestUtils from 'react-addons-test-utils';

let renderer;

describe('EventResult DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });
   it('renders correctly homeScore > awayScore', () => {
      expect(renderer.render(
         // never forget
         <EventResult
            homeName='Brazil'
            awayName='Germany'
            homeScore={1}
            awayScore={7}
         />
      )).toMatchSnapshot();
   });

   it('renders correctly homeScore < awayScore', () => {
      expect(renderer.render(
         <EventResult
            homeName='Sweden'
            awayName='Italy'
            homeScore={4}
            awayScore={1}
         />
      )).toMatchSnapshot();
   });

   it('renders correctly homeScore == awayScore', () => {
      expect(renderer.render(
         <EventResult
            homeName='France'
            awayName='England'
            homeScore={2}
            awayScore={2}
         />
      )).toMatchSnapshot();
   });
});
