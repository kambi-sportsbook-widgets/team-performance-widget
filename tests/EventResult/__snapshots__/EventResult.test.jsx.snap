// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventResult DOM rendering renders correctly homeScore < awayScore 1`] = `
<div
  className="kw-eventresult"
>
  <div
    className="kw-eventresult__participant KambiWidget-card-text-color"
  >
    <div
      className="kw-eventresult__participant__name"
    >
      Sweden
    </div>
    <div
      className="kw-eventresult__participant__score"
    >
      4
    </div>
  </div>
  <div
    className="kw-eventresult__participant KambiWidget-card-support-text-color"
  >
    <div
      className="kw-eventresult__participant__name"
    >
      Italy
    </div>
    <div
      className="kw-eventresult__participant__score"
    >
      1
    </div>
  </div>
</div>
`;

exports[`EventResult DOM rendering renders correctly homeScore == awayScore 1`] = `
<div
  className="kw-eventresult"
>
  <div
    className="kw-eventresult__participant KambiWidget-card-support-text-color"
  >
    <div
      className="kw-eventresult__participant__name"
    >
      France
    </div>
    <div
      className="kw-eventresult__participant__score"
    >
      2
    </div>
  </div>
  <div
    className="kw-eventresult__participant KambiWidget-card-support-text-color"
  >
    <div
      className="kw-eventresult__participant__name"
    >
      England
    </div>
    <div
      className="kw-eventresult__participant__score"
    >
      2
    </div>
  </div>
</div>
`;

exports[`EventResult DOM rendering renders correctly homeScore > awayScore 1`] = `
<div
  className="kw-eventresult"
>
  <div
    className="kw-eventresult__participant KambiWidget-card-support-text-color"
  >
    <div
      className="kw-eventresult__participant__name"
    >
      Brazil
    </div>
    <div
      className="kw-eventresult__participant__score"
    >
      1
    </div>
  </div>
  <div
    className="kw-eventresult__participant KambiWidget-card-text-color"
  >
    <div
      className="kw-eventresult__participant__name"
    >
      Germany
    </div>
    <div
      className="kw-eventresult__participant__score"
    >
      7
    </div>
  </div>
</div>
`;
